### ISSUE TITLE
Renaming default pipeline and disconnecting power corrupts settings

### ISSUE THREAD

**Issue by CJKeller03** [Chameleon-Vision/chameleon-vision#63]
Hello, while running the latest version of Chameleon on a Raspberry Pi 4, I ran into an unusual issue.

Because of the way the RPi is connected to our robot, every time the robot is shut off, the RPi loses power without going through the normal shut down sequence. If the default pipeline has not been renamed,  when the RPi is booted back up Chameleon starts automatically (using a script in rc.local) and works normally, with all camera settings retained.

However, if the default pipeline is renamed and this occurs, Chameleon fails to start correctly on boot, and attempting to start it from the command line results in failure to connect to the camera. This happens regardless of whether the save button in the web interface was pressed, or how long it has been since the pipeline was renamed. The only way (that I could find) to resolve the issue is to delete the /boot/chameleon-vision folder on the RPi and restart Chameleon, which deletes all camera settings.

I can send pictures of the terminal output when this occurs, but currently don't have access to the RPi. Thanks for your help!

**Comment by prateekma** [comment]
A quick fix for this would be appreciated as competitions will be starting in Week 1. If the program fails to run between playoff matches, it would be a huge headache and would probably prompt us to switch to a different solution.

**Comment by CJKeller03** [comment]
> A quick fix for this would be appreciated as competitions will be starting in Week 1.

The temporary solution I found was to just use the default pipeline without renaming it. Like I said before, Chameleon works fine if the pipeline hasn't been renamed.

**Comment by Bankst** [comment]
> 
> The temporary solution I found was to just use the default pipeline without renaming it. Like I said before, Chameleon works fine if the pipeline hasn't been renamed.

If you create a second pipeline and rename *that* one, what happens? If this is specific to renaming the default pipeline (`New Pipeline`) then it may be an easy fix.

**Comment by CJKeller03** [comment]
While I was setting up a backup RPi with chameleon, I tried to reproduce the issue on it. When I just renamed any of the pipelines without changing any of the other settings and disconnected power to the Pi, nothing happened. I couldn't reproduce the issue until after I set up the static IP and hostname too. Here is the output from the terminal:

pi@raspberrypi:~ $ sudo java -jar chameleon-vision.jar 
Starting Chameleon Vision on platform Linux Raspbian
Settings folder: /boot/chameleon-vision/settings
No valid network interfaces found! Staying unmanaged.
Starting NT Client
VIDEOIO ERROR: V4L2: Could not obtain specifics of capture window.
VIDEOIO ERROR: V4L: can't open camera by index 1
/dev/video1 does not support memory mapping
VIDEOIO ERROR: V4L2: Unable to capture video memory.VIDEOIO ERROR: V4L: can't open camera by index 10
/dev/video10 does not support memory mapping
VIDEOIO ERROR: V4L2: Unable to capture video memory.VIDEOIO ERROR: V4L: can't open camera by index 11
/dev/video11 does not support memory mapping
VIDEOIO ERROR: V4L2: Unable to capture video memory.VIDEOIO ERROR: V4L: can't open camera by index 12
/dev/video12 does not support memory mapping
Exception in thread "main" java.lang.NullPointerException
	at com.chameleonvision.vision.camera.USBCameraCapture.<init>(USBCameraCapture.java:31)
	at com.chameleonvision.vision.VisionManager.initializeProcesses(VisionManager.java:87)
	at com.chameleonvision.Main.main(Main.java:166)
NT Connection has failed! Will retry in background.

When I opened the webserver in Chromium, the FPS was NaN, the pipeline and camera boxes were blank, and the settings menu was inaccessible. I took a picture of it, so I'll send that as soon as I can.

The same issue occurrs when I rename one of the other pipelines.
